framework/include/meshgenerators/MeshGenerator.h

```diff
@@ -213,6 +327,212 @@ MeshGenerator::declareMeshProperty(const std::string & data_name, const T & init
+      _type_names.begin(), std::find(_type_names.begin(), _type_names.end(), old_type_str));
+
+  switch (type_id)
+  {
+    case 0:
+      declareMeshProperty<std::vector<unsigned int>>(data_name);
+      break;
+    case 1:
+      declareMeshProperty<double>(data_name);
+      break;
+    case 2:
+      declareMeshProperty<std::vector<double>>(data_name);
+      break;
+    case 3:
+      declareMeshProperty<unsigned short>(data_name);
+      break;
+    case 4:
+      declareMeshProperty<unsigned int>(data_name);
+      break;
+    case 5:
+      declareMeshProperty<bool>(data_name);
+      break;
+    case 6:
+      declareMeshProperty<unsigned long long>(data_name);
+      break;
+    case 7:
+      declareMeshProperty<std::string>(data_name);
+    case 8:
+      declareMeshProperty<int>(data_name);
+      break;
+    case 9:
+      declareMeshProperty<Point>(data_name);
+      break;
+    case 10:
+      declareMeshProperty<std::vector<unsigned int>>(data_name);
+      break;
+    case 11:
+      declareMeshProperty<std::vector<int>>(data_name);
+      break;
+    case 12:
+      declareMeshProperty<std::vector<unsigned short>>(data_name);
+      break;
+    case 13:
+      declareMeshProperty<std::vector<unsigned long long>>(data_name);
+      break;
+    case 14:
+      declareMeshProperty<std::vector<Point>>(data_name);
+      break;
+    case 15:
+      declareMeshProperty<std::vector<std::vector<double>>>(data_name);
+      break;
+    case 16:
+      declareMeshProperty<std::map<std::string, std::pair<unsigned short, unsigned long long>>>(
+          data_name);
+      break;
+    case 17:
+      declareMeshProperty<short>(data_name);
+      break;
+    case 18:
+      declareMeshProperty<std::map<subdomain_id_type, std::vector<std::vector<subdomain_id_type>>>>(
+          data_name);
+      break;
+    case 19:
+      declareMeshProperty<std::map<subdomain_id_type, std::vector<std::vector<std::string>>>>(
+          data_name);
+      break;
+    case 20:
+      declareMeshProperty<std::vector<std::string>>(data_name);
+      break;
+    case 21:
+      declareMeshProperty<std::vector<std::vector<short>>>(data_name);
+      break;
+    case 22:
+      declareMeshProperty<std::vector<std::vector<std::string>>>(data_name);
+      break;
+    case 23:
+      declareMeshProperty<std::map<BoundaryID, RealVectorValue>>(data_name);
+      break;
+    default:
+      mooseError("In Mesh Generator ",
+                 _name,
+                 ": the forwarded mesh metadata named ",
+                 data_name,
+                 " has the type, ",
+                 old_type_str,
+                 ", that has not been included in _type_names.");
+  }
+  return type_id;
+}
+
+inline void
+MeshGenerator::setForwardedMeshProperty(const std::string data_name,
+                                        const std::string input_mg_name,
+                                        const unsigned int type_id)
+{
+  switch (type_id)
+  {
+    case 0:
```

Comment:
this is prone to error. Can we switch on the name instead?

Reply:
We cannot switch on strings. Maybe use enum?

Reply:
Or I can make a function for this.

Reply:
enum seems fine too. Just not mapping manually from a string to an int